app/src/main/java/org/schabi/newpipe/settings/NewPipeSettings.java

```diff
@@ -76,6 +76,14 @@ public static void initSettings(final Context context) {
 
         saveDefaultVideoDownloadDirectory(context);
         saveDefaultAudioDownloadDirectory(context);
+
+
+        final int lastMediaTunnelingUpdate = PreferenceManager.getDefaultSharedPreferences(context)
+                .getInt(context.getString(R.string.media_tunneling_device_blacklist_version), 0);
+        if (Boolean.TRUE.equals(isFirstRun) || DeviceUtils.MEDIA_TUNNELING_DEVICE_BLACKLIST_VERSION
+                != lastMediaTunnelingUpdate) {
+            setMediaTunneling(context);
```

Comment:
Shouldn't you also check whether it was turned off manually before re-enabling tunneling?

Reply:
done